src/main/webapp/app/assessment-shared/assessment-shared.module.ts

```diff
@@ -22,8 +25,18 @@ import { AssessmentFiltersComponent } from 'app/assessment-shared/assessment-fil
         ScoreDisplayComponent,
         AssessmentDetailComponent,
         AssessmentFiltersComponent,
+        ExternalSubmissionButtonComponent,
+        ExternalSubmissionDialogComponent,
     ],
-    exports: [AssessmentLayoutComponent, AssessmentGeneralFeedbackComponent, ScoreDisplayComponent, AssessmentDetailComponent, AssessmentFiltersComponent],
-    providers: [JhiAlertService],
+    entryComponents: [ExternalSubmissionDialogComponent],
+    exports: [
+        AssessmentLayoutComponent,
+        AssessmentGeneralFeedbackComponent,
+        ScoreDisplayComponent,
+        AssessmentDetailComponent,
+        AssessmentFiltersComponent,
+        ExternalSubmissionButtonComponent,
+    ],
+    providers: [JhiAlertService, ExampleSubmissionService],
```

Comment:
```suggestion
    providers: [JhiAlertService],
```
If you specify `providedIn: 'root'` in your service, then Angular will take care of injecting it where needed. You should almost never specify services in modules, since this can break your app, because of problems like circular dependencies (We also had some really annoying problems because of that, just like with the barrel files).
You also prevent Angular from performing tree shaking, if you specify a service like this.

Reply:
Thanks, very good point. I removed the explicit service import here.